File: Baekjoon/yerim/회의실 배정.cpp
// [1931] 회의실 배정
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;

int n;
vector<pair<int, int> > meeting;

void input()
{
    int start, end;
    cin >> n;
    for(int i = 0 ; i < n ; i++) {
        cin >> start >> end;
        meeting.push_back(make_pair(end, start));
    }
}

int maxNum()
{
    int cnt = 0, endTime = 0;
    sort(meeting.begin(), meeting.end());
    for(int i = 0 ; i < n ; i++) {
        if(meeting[i].second >= endTime) { // 다음 회의의 시작 시간 >= 현재 회의의 종료 시간
            cnt++;
            endTime = meeting[i].first;
        }
    }
    return cnt;
}

int main(void)
{
    cin.tie(NULL); 
    ios_base::sync_with_stdio(false);
    
    input();
    cout << maxNum() << '\n';
    
    return 0;
}

File: Baekjoon/subin/회의실 배정.py
n = int(input())
meeting = [list(map(int, input().split())) for _ in range(n)]
meeting.sort(key=lambda x: (x[1], x[0]))

count = 1
start, end = meeting[0]

for i in range(1, n):
    if end <= meeting[i][0]:
        count += 1
        end = meeting[i][1]

print(count)